docs/5.0-breaking-changes/nest-breaking-changes.md

```diff
@@ -0,0 +1,3410 @@
+
+2.x: `public IDisposable Subscribe(IObserver<IReindexResponse<T>> observer)`  
+5.x: `public IDisposable Subscribe(ReindexObserver<T> observer)`  
+
+**public method Nest.ReindexObserver&lt;T&gt;..ctor** *Declaration changed (Breaking)*
+
+2.x: `public  .ctor(Action<IReindexResponse<T>> onNext, Action<Exception> onError, Action completed)`  
+5.x: `public  .ctor(Action<IBulkAllResponse> onNext, Action<Exception> onError, Action onCompleted)`  
+
+**public method Nest.ReindexOnServerDescriptor.RequestsPerSecond** *Declaration changed (Breaking)*
+
+2.x: `public ReindexOnServerDescriptor RequestsPerSecond(Single requests_per_second)`  
+5.x: `public ReindexOnServerDescriptor RequestsPerSecond(long requests_per_second)`  
+
+**public property Nest.ReindexOnServerRequest.RequestsPerSecond** *Declaration changed (Breaking)*
+
+2.x: `public Single RequestsPerSecond { get; set; }`  
+5.x: `public long RequestsPerSecond { get; set; }`  
+
+**public method Nest.ReindexRethrottleDescriptor.RequestsPerSecond** *Declaration changed (Breaking)*
+
+2.x: `public ReindexRethrottleDescriptor RequestsPerSecond(Single requests_per_second)`  
+5.x: `public ReindexRethrottleDescriptor RequestsPerSecond(long requests_per_second)`  
+
+**public property Nest.ReindexRethrottleRequest.RequestsPerSecond** *Declaration changed (Breaking)*
+
+2.x: `public Single RequestsPerSecond { get; set; }`  
+5.x: `public long RequestsPerSecond { get; set; }`  
+
+**public property Nest.Time.Milliseconds** *Declaration changed (Breaking)*
+
+2.x: `public double Milliseconds { get; private set; }`  
+5.x: `public double? Milliseconds { get; private set; }`  
+
+**public method Nest.UpdateByQueryDescriptor&lt;T&gt;.RequestsPerSecond** *Declaration changed (Breaking)*
+
+2.x: `public UpdateByQueryDescriptor<T> RequestsPerSecond(Single requests_per_second)`  
+5.x: `public UpdateByQueryDescriptor<T> RequestsPerSecond(long requests_per_second)`  
+
+**public property Nest.UpdateByQueryRequest.RequestsPerSecond** *Declaration changed (Breaking)*
+
+2.x: `public Single RequestsPerSecond { get; set; }`  
+5.x: `public long RequestsPerSecond { get; set; }`  
+
+**public property Nest.UpdateByQueryRequest&lt;T&gt;.RequestsPerSecond** *Declaration changed (Breaking)*
+
+2.x: `public Single RequestsPerSecond { get; set; }`  
+5.x: `public long RequestsPerSecond { get; set; }`  
+
+**public property Nest.Watch.Actions** *Declaration changed (Breaking)*
+
+2.x
+```csharp
+[JsonPropertyAttribute("actions")]
+[JsonConverterAttribute(Nest.ActionsJsonConverter)]
+public IDictionary<string, IAction> Actions { get; internal set; }
+```
+
+5.x
+```csharp
+[JsonPropertyAttribute("actions")]
+[JsonConverterAttribute(Nest.ActionsJsonConverter)]
+public Actions Actions { get; internal set; }
+```
+
+# Read only data types on responses.
+
+Responses now favor `IReadOnlyDictionary` and `IReadOnlyCollection` which are initialized as empty.
```

Comment:
Use the open generic type names `IReadOnlyDictionary<TKey, TValue>` and `IReadOnlyCollection<T>`